src/posix/client.cpp

```diff
@@ -182,22 +182,56 @@ int main(int argc, char *argv[])
         char   buffer[kLineBufferSize];
         size_t count = 0;
 
-        for (int i = 1; i < argc; i++)
+        enum
         {
-            for (const char *c = argv[i]; *c; ++c)
+            kStateArgument,
+            kStateEscaping,
+            kStateEnd,
+        } state = kStateArgument;
+
+        for (char **arg = &argv[1]; --argc && count < sizeof(buffer); ++arg)
+        {
+            switch (state)
             {
-                if (IsSeparator(*c))
+            case kStateEnd:
+                buffer[count++] = ' ';
+                // fall through
+
+            case kStateArgument:
+                for (const char *c = *arg; *c && count < sizeof(buffer);)
                 {
-                    buffer[count++] = '\\';
+                    if (state == kStateEscaping)
+                    {
+                        state = kStateArgument;
+                    }
+                    else if (IsSeparator(*c))
+                    {
+                        state           = kStateEscaping;
```

Comment:
If the last character is `\`, the current behavior is incorrect because it will escape the next separator. 

I think we should treat this `\` as `\\`.